framework/src/controller/bus.js

```diff
@@ -178,17 +183,46 @@ class Bus extends EventEmitter2 {
 			throw new Error(`Event ${eventName} is not registered to bus.`);
 		}
 
-		super.emit(eventName, eventValue); // Communicate throw event emitter
-		this.pubSocket.emit(eventName, eventValue); // Communicate throw unix socket
+		// Communicate through event emitter
+		super.emit(eventName, eventValue);
+
+		// Communicate through unix socket
+		if (this.config.ipc.enabled) {
+			this.pubSocket.emit(eventName, eventValue);
+		}
 	}
 
 	subscribe(eventName, cb) {
 		if (!this.getEvents().includes(eventName)) {
-			throw new Error(`Event ${eventName} is not registered to bus.`);
+			this.logger.info(
+				`Event ${eventName} was subscribed but not registered to the bus yet.`
+			);
+		}
+
+		// Communicate through event emitter
+		super.on(eventName, cb);
```

Comment:
`this.on`?

Reply:
I kind of prefer to use `super` here because it's instantly clear that we are using a parent class method. But I don't have strong opinion on this.